tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -163,39 +211,119 @@ void test_boundary_correction_impl(
+  using face_tags_with_curved_background = tmpl::conditional_t<
+      curved_background,
+      tmpl::remove_duplicates<tmpl::push_back<
+          face_tags, typename detail::inverse_spatial_metric_tag<
+                         curved_background>::template f<System>>>,
+      face_tags>;
+
   Variables<dg_package_field_tags> interior_package_data{used_for_size.size()};
-  const auto interior_fields_on_face = make_with_random_values<
-      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
-                             package_primitive_tags>>>(
-      make_not_null(&gen), make_not_null(&dist), used_for_size);
+  const auto interior_fields_on_face =
+      make_with_random_values<Variables<face_tags_with_curved_background>>(
+          make_not_null(&gen), make_not_null(&dist), used_for_size);
 
   Variables<dg_package_field_tags> exterior_package_data{used_for_size.size()};
-  const auto exterior_fields_on_face = make_with_random_values<
-      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
-                             package_primitive_tags>>>(
-      make_not_null(&gen), make_not_null(&dist), used_for_size);
+  const auto exterior_fields_on_face =
+      make_with_random_values<Variables<face_tags_with_curved_background>>(
+          make_not_null(&gen), make_not_null(&dist), used_for_size);
 
   // Compute the interior and exterior normal vectors so they are pointing in
   // opposite directions.
   auto interior_unit_normal_covector = make_with_random_values<
       tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
       make_not_null(&gen), make_not_null(&pos_neg_dist), used_for_size);
-  const Scalar<DataVector> interior_normal_magnitude =
-      magnitude(interior_unit_normal_covector);
-  for (auto& t : interior_unit_normal_covector) {
-    t /= get(interior_normal_magnitude);
+  tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>
+      interior_unit_normal_vector{};
+  if constexpr (not curved_background) {
+    const Scalar<DataVector> interior_normal_magnitude =
+        magnitude(interior_unit_normal_covector);
+    for (auto& t : interior_unit_normal_covector) {
+      t /= get(interior_normal_magnitude);
+    }
+  } else {
+    using inv_spatial_metric = typename detail::inverse_spatial_metric_tag<
+        curved_background>::template f<System>;
+    // Make sure the inverse spatial metric is reasonable.
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      for (size_t j = i; j < FaceDim + 1; ++j) {
+        // NOLINTNEXTLINE
+        const_cast<DataVector&>(
+            get<inv_spatial_metric>(interior_fields_on_face).get(i, j)) *= 0.01;
+      }
+    }
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      // NOLINTNEXTLINE
+      const_cast<DataVector&>(
+          get<inv_spatial_metric>(interior_fields_on_face).get(i, i)) += 1.0;
+    }
+
+    // make the exterior inverse spatial metric be close to the interior one
+    // since the solution should be smooth.
```

Comment:
But this is so close! Would a slightly larger difference make debugging easier? (honest question, not a change request)

Reply:
Increase to 0.999, I don't want to go larger because we want to keep the diagonals dominant. I've added a comment stating that.